app/scripts/metamask-controller.js

```diff
@@ -1647,15 +1647,16 @@ export default class MetamaskController extends EventEmitter {
   /**
    * Gets network state relevant for external providers.
    *
-   * @param {object} [memState] - The MetaMask memState. If not provided,
-   * this function will retrieve the most recent state.
    * @returns {object} An object with relevant network state properties.
    */
-  getProviderNetworkState(memState) {
-    const { network } = memState || this.getState();
+  getProviderNetworkState() {
+    const { provider, networkStatus } = this.networkController.store.getState();
     return {
-      chainId: this.networkController.store.getState().provider.chainId,
-      networkVersion: network,
+      chainId: provider.chainId,
+      // NOTE: For compatibility with providers, this must return "loading" if
```

Comment:
I found this out when running the e2e tests that interact with the test dapp. It'd be nice if we could remove `networkVersion` altogether from this object but we'd have to update `@metamask/providers` (and it would probably impact dapps).